TombEngine/Objects/Generic/Switches/turn_switch.cpp

```diff
@@ -178,40 +181,40 @@ namespace TEN::Entities::Switches
 
 		if (g_Level.Items[itemNumber].ItemFlags[0] == TURN_SWITCH_CLOCKWISE)
 		{
-			if (switchItem->Animation.AnimNumber == Objects[switchItem->ObjectNumber].animIndex + 2)
+			if (switchItem->Animation.AnimNumber == 2)
 			{
 				switchItem->Pose.Orientation.y += ANGLE(90.0f);
 				switchItem->DisableInterpolation = true;
 
 				if (IsHeld(In::Action))
 				{
 					laraItem->Animation.AnimNumber = LA_TURNSWITCH_PUSH_CLOCKWISE_START;
-					laraItem->Animation.FrameNumber = GetAnimData(laraItem).frameBase;
+					laraItem->Animation.FrameNumber = 0;
 
-					switchItem->Animation.AnimNumber = Objects[switchItem->ObjectNumber].animIndex + 1;
-					switchItem->Animation.FrameNumber = GetAnimData(switchItem).frameBase;
+					switchItem->Animation.AnimNumber = 1;
+					switchItem->Animation.FrameNumber = 0;
 				}
 			}
 
 			if (laraItem->Animation.AnimNumber == LA_TURNSWITCH_PUSH_CLOCKWISE_END &&
-				laraItem->Animation.FrameNumber == GetAnimData(laraItem).frameEnd &&
+				TestLastFrame(*laraItem) && 
 				!switchItem->ItemFlags[1])
 			{
 				switchItem->ItemFlags[1] = 1;
 				switchItem->DisableInterpolation = true;
 			}
 
-			if ((laraItem->Animation.FrameNumber >= GetAnimData(*laraItem, LA_TURNSWITCH_PUSH_CLOCKWISE_START).frameBase &&
-				laraItem->Animation.FrameNumber <= GetAnimData(*laraItem, LA_TURNSWITCH_PUSH_CLOCKWISE_START).frameBase + 43) ||
-				(laraItem->Animation.FrameNumber >= GetAnimData(*laraItem, LA_TURNSWITCH_PUSH_CLOCKWISE_START).frameBase + 58 &&
-				laraItem->Animation.FrameNumber <= GetAnimData(*laraItem, LA_TURNSWITCH_PUSH_CLOCKWISE_START).frameBase + 115))
+			if ((laraItem->Animation.FrameNumber >= 0 &&
```

Comment:
I believe we also need to check if animation is `LA_TURNSWITCH_PUSH_CLOCKWISE_START` here?